src/riak_kv_wm_queue.erl

```diff
@@ -77,25 +93,100 @@ service_available(RD, Ctx=#ctx{riak=RiakProps}) ->
              Ctx}
     end.
 
+
+allowed_methods(RD, Ctx) ->
+    {['GET', 'POST'], RD, Ctx}.
+
 -spec malformed_request(#wm_reqdata{}, context()) ->
     {boolean(), #wm_reqdata{}, context()}.
 %% @doc Determine whether request is well-formed
-malformed_request(RD, Ctx) ->
-    QueueNameRaw = wrq:path_info(queuename, RD),
-    QueueName =
-        list_to_atom(riak_kv_wm_utils:maybe_decode_uri(RD, QueueNameRaw)),
+malformed_request(RD, Ctx) when Ctx#ctx.method =:= 'GET' ->
+    QueueName = malformed_queuename(RD),
     ObjectFormatRaw = wrq:get_qs_value(?Q_OBJECT_FORMAT, "internal", RD),
     ObjectFormat = list_to_atom(ObjectFormatRaw),
-    {false, RD, Ctx#ctx{queuename = QueueName, object_format = ObjectFormat}}.
+    {false, RD, Ctx#ctx{queuename = QueueName, object_format = ObjectFormat}};
+malformed_request(RD, Ctx) when Ctx#ctx.method =:= 'POST' ->
+    QueueName = malformed_queuename(RD),
+    KeyClockList =
```

Comment:
Same here... this kind of crashes or always succeeds

Reply:
Now errors